app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModel.kt

```diff
@@ -219,12 +240,14 @@ class ConnectionsListViewModel(
         }
     }
 
-    fun onDialogActionIdClick(dialogActionId: Int, actionsGoToSettings: Int) {
+    fun onDialogActionIdClick(dialogActionId: Int, actionResId: ResId, guid: GUID = "") {
         if (dialogActionId == DialogInterface.BUTTON_POSITIVE) {
-            if (actionsGoToSettings == R.string.actions_proceed) {
-                onAskPermissionsEvent.postUnitEvent()
-            } else if (actionsGoToSettings == R.string.actions_go_to_settings) {
-                onGoToSettingsEvent.postUnitEvent()
+            when (actionResId) {
+                R.string.actions_proceed -> onAskPermissionsEvent.postUnitEvent()
+                R.string.actions_go_to_settings -> onGoToSettingsEvent.postUnitEvent()
+                R.string.actions_retry -> {
+                    if (!noInternetConnection) onDeleteClickEvent.postValue(ViewModelEvent(guid))
```

Comment:
not defined else case

Reply:
Why you show new dialog in case when was pressed actions_retry and internet connection is good?

Reply:
discussed